src/components/views/dialogs/ResetIdentityDialog.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+import React, { type JSX, type MouseEventHandler } from "react";
+
+import { MatrixClientPeg } from "../../../MatrixClientPeg";
+import MatrixClientContext from "../../../contexts/MatrixClientContext";
+import { ResetIdentityBody, type ResetIdentityBodyVariant } from "../settings/encryption/ResetIdentityBody";
+
+interface ResetIdentityDialogProps {
+    /**
+     * Called when the dialog closes.
+     */
+    onFinished: () => void;
+
+    /**
+     * Called when the identity is reset (before onFinished is called).
+     */
+    onReset: MouseEventHandler<HTMLButtonElement>;
```

Comment:
It seems a bit weird for 'onReset' to take a mouse event: it's hard to see how the original click would be relevant.

Reply:
It's just copied from the original settings panel (where I don't think the mouse event is used either).  We can get rid of it here.

Reply:
I've changed this to be a function with no arguments.  Should I also change `ResetIdentityBody` and `ResetIdentityPanel` (since we already touch those lines in this PR)?

Reply:
Oh I see - thanks. Up to you whether you fancy changing it elsewhere.